labelbox_dev/data_row.py

```diff
@@ -179,9 +184,59 @@ def get_by_global_keys(
                 raise LabelboxError(
                     f"Failed to retrieve task information for `data_row.get_by_global_keys()` async operation"
                 )
-            return BulkGetDataRowsTask.get_by_id(task_id)
+            return GetDataRowsTask.get_by_id(task_id)
 
         return IdentifierPaginator(f"{DATA_ROW_RESOURCE}",
                                    DataRow,
                                    global_keys,
                                    identifiers_key='global_keys')
+
+    @staticmethod
+    def update_many(data_rows: List[UpdateDataRowsWithIdType],
+                    run_async: bool = False):
+        for data_row in data_rows:
+            if 'row_data' in data_row:
+                data_row = DataRow._format_data_rows([data_row])[0]
+
+        body = {'data_rows': data_rows, 'run_async': run_async}
+        if run_async:
+            task_json = Session.post_request(f"{DATA_ROW_RESOURCE}/bulkupdate",
+                                             json=body)
+            task_id = task_json.get(
+                'taskId')  # TODO: change to snake_case from backend
+
+            if task_id is None:
+                raise LabelboxError(
+                    f"Failed to retrieve task information for `data_row.update_many()` async operation"
+                )
+            return UpdateDataRowsTask.get_by_id(task_id)
+
+        data_rows = Session.post_request(f"{DATA_ROW_RESOURCE}/bulkupdate",
+                                         json=body)
+        return [DataRow(data_row_json) for data_row_json in data_rows]
+
+    @staticmethod
+    def delete_many(ids: Optional[List[str]] = None,
+                    global_keys: Optional[List[str]] = None,
+                    run_async: bool = False):
+        body = {'run_async': run_async}
+        if ids:
+            body.update({'ids': ids})
+        if global_keys:
+            body.update({'global_keys': global_keys})
+
+        if run_async:
+            task_json = Session.post_request(f"{DATA_ROW_RESOURCE}/bulkdelete",
+                                             json=body)
+            task_id = task_json.get(
+                'taskId')  # TODO: change to snake_case from backend
+
+            if task_id is None:
+                raise LabelboxError(
+                    f"Failed to retrieve task information for `data_row.delete_many()` async operation"
+                )
+            return DeleteDataRowsTask.get_by_id(task_id)
+
+        data_rows = Session.post_request(f"{DATA_ROW_RESOURCE}/bulkdelete",
```

Comment:
can we call this and the one above just once before if, i.e. 
```
data_rows = Session.post_request(f"{DATA_ROW_RESOURCE}/bulkdelete",
...
if run_async:
...
```

Reply:
Ah that's much cleaner. If i end up putting both sync and async into one method, I will clean this up